app/workers/stale_issue_marker.rb

```diff
@@ -23,49 +23,50 @@ class StaleIssueMarker
 Thank you for all your contributions!
   EOS
 
-  attr_reader :fq_repo_name
-
-  def perform(fq_repo_name)
-    @fq_repo_name = fq_repo_name
-    raise "The label #{STALE_LABEL_NAME} does not exist on #{fq_repo_name}" unless GithubService.valid_label?(fq_repo_name, STALE_LABEL_NAME)
-
-    GithubService.issues(fq_repo_name, :state => :open, :sort => :updated, :direction => :asc).each do |issue|
-      pinned = (issue.labels & PINNED_LABELS).present?
-
-      if issue.updated_at < stale_date && !pinned
-        if issue.pull_request?
-          closable_prs << issue
-        else
-          stale_issues << issue
-        end
+  def perform
+    GithubService.search_issues(*search_args).each do |issue|
+      if issue.pull_request?
+        close_pr(issue)
+      else
+        mark_as_stale(issue)
       end
     end
-
-    stale_issues.each do |issue|
-      next if issue.labels.include?(STALE_LABEL_NAME)
-      logger.info("[#{Time.now.utc}] - Marking issue #{fq_repo_name}##{issue.number} as stale")
-      issue.add_labels([STALE_LABEL_NAME])
-      issue.add_comment(STALE_ISSUE_MESSAGE)
-    end
-
-    closable_prs.each do |pr|
-      logger.info("[#{Time.now.utc}] - Closing stale PR #{fq_repo_name}##{pr.number}")
-      GithubService.close_pull_request(pr.fq_repo_name, pr.number)
-      pr.add_comment(CLOSABLE_PR_MESSAGE)
-    end
   end
 
   private
 
+  def search_args
+    query  = "is:open archived:false update:<#{stale_date.strftime('%Y-%m-%d')}"
+    query << " #{PINNED_LABELS.map { |label| %(-label:"#{label}") }.join(" ")}"
```

Comment:
You can probably also remove already stale issues with `-label:stale`

Reply:
I decided against doing that since this is adding `stale` as part of the query, since we will be closing some of them in the future, based on this:

https://github.com/ManageIQ/miq_bot/issues/390#issuecomment-577715811

In the end, the labels get included with the results, so there isn't an extra query that is being called by doing this, it is just trimming down the results a bit.